### DISCUSSION TITLE
What's the difference between all plugins under 1 use {} call vs each having their own?

### DISCUSSION THREAD

**Discussion by Corey-Keller** [wbthomason/packer.nvim#573] [Q&A] [answered]
Are there any differences in how they're handled? Or pros/cons to either approach? 

```
return require("packer").startup({function(use) use {
	-- Packer can manage itself as an optional plugin
	{ "wbthomason/packer.nvim", opt = true },

	{
		"folke/which-key.nvim",
		config = function ()
			require('which-key').setup()
			--require('plugins.which-key')
		end,
	},

...

} end
```

vs

```
return require("packer").startup({function(use)
	-- Packer can manage itself as an optional plugin
	use({ "wbthomason/packer.nvim", opt = true })

	use {
		"folke/which-key.nvim",
		config = function ()
			require('which-key').setup()
			--require('plugins.which-key')
		end,
	}

...

end
```

**Comment by wbthomason** [comment, marked as answer]
Sorry for my delayed response; I was away due to work (#562). There's no real semantic difference to these two styles; the former is mostly used if you want to group a subset of related plugins together in your specification file, and arises in part because of handling of nested specifications with `requires`.

**Reply by Corey-Keller** [reply]
First, Thanks for your help, and your awesome plugin manager.

Second, I love how you communicated your absence with a GitHub issue! I'd have never considered that, and it seems really cool to me.

Third, I get that there's no real semantic difference, but is there any performance difference in any situations? That's actually what prompted this question: The update checks done during `PackerSync` _seem_ to be a bit slower when it's all nested under 1 use call vs each plugin getting its own, but I'm 100% aware that it could just be a placebo effect, so I wanted to ask. :)

**Reply by wbthomason** [reply]
There definitely shouldn't be any performance difference - under the hood, the nested specifications get flattened out before any plugin operations occur, so I suspect this is just a nocebo (negative placebo). That said, if you can reproducibly show a relative slowdown for either form of specification, I'd be super interested to hear about it!
